BAZProject/Modules/Home/View/HomeViewController.swift

```diff
@@ -0,0 +1,209 @@
+        getData()
+    }
+
+    private func getData() {
+        presenter?.willFetchMovieTopRated()
+        presenter?.willFetchNowPlaying()
+        presenter?.willFetchPopularMovies()
+        presenter?.willFetchUpcomingMovies()
+    }
+
+    private func addObservers() {
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(aumentMovieShow),
+                                               name: .notificacionCenterNameShowDetail,
+                                               object: nil)
+    }
+
+    @objc private func aumentMovieShow(_ notification: Notification) {
+        totalMoviesShow += LocalizedConstants.commonIncrementNumber
+    }
+}
+
+extension HomeViewController: HomeViewProtocol {
+    func updateView(data: [UpcomingModelResult]) {
+        upcomingMovies = data
+        var posterUrlString: [String] = []
+        data.forEach { movie in
+            if let poster = movie.posterPath {
+                posterUrlString.append(poster)
+            }
+        }
+
+        upcomingMoviesSlider.setUp(imageUrlArray: posterUrlString, imageContentMode: .scaleAspectFit)
+        upcomingMoviesSlider.delegate = self
+    }
+
+    func updateView(data: [PopularMoviesModelResult]) {
+        popularMovies = data
+        var posterUrlString: [String] = []
+        data.forEach { movie in
+            if let poster = movie.posterPath {
+                posterUrlString.append(poster)
+            }
+        }
+
+        popularImageSlider.setUp(imageUrlArray: posterUrlString, imageContentMode: .scaleAspectFit)
+        popularImageSlider.delegate = self
+    }
+
+    func updateView(data: [NowPlayingResult]) {
+        nowPlaying = data
+        var posterUrlString: [String] = []
+        data.forEach { movie in
+            if let poster = movie.posterPath {
+                posterUrlString.append(poster)
+            }
+        }
+
+        nowPlayingSlider.setUp(imageUrlArray: posterUrlString, imageContentMode: .scaleAspectFit)
+        nowPlayingSlider.delegate = self
+    }
+
+    func updateView(data: [MovieTopRatedResult]) {
+        movieTopRated = data
+        var cellMovieType = [CellMovieType]()
+        data.forEach { movie in
+            if let bac = movie.backdropPath {
+                let imageUrlString: String = Endpoint.img(idImage: bac, sizeImage: .w500).urlString
+                cellMovieType.append(CellMovieType(imageUrlString: imageUrlString, title: movie.title ?? ""))
+            }
+        }
+
+        movieTopSlider.setUp(cellMovieType)
+        movieTopSlider.delegate = self
+    }
+
+    func stopLoading() {
+        guaranteeMainThread {
+            self.isLoading = false
+            self.view.removeLoader()
+        }
+    }
+
+    func setErrorGettingData(_ status: Bool) {
+        errorGetData = status
+    }
+}
+
+extension HomeViewController: ImageSliderDelegate {
+    func indexDidSelect(_ index: Int, object: ImageSlider) {
```

Comment:
Puedes intentar explorar el patrón de diseño [Chain Of Responsibility](https://refactoring.guru/design-patterns/chain-of-responsibility), para evitar caer en cascadas de `else if`.

Otro patrón de diseño que puede funcionar, pero ya sin implementar el Chain of Responsibility, sería el [Abstract Factory](https://refactoring.guru/design-patterns/abstract-factory).

Si te animas a probarlos, en un principio sería más sencillo empezar con el Chain, que únicamente sería hacer un refactor sobre los if elses.

En cambio el Abstract Factory te llevaría a crear tus sliders con un enfoque muy distinto.